coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala

```diff
@@ -53,9 +65,10 @@ object LogicalPlanUtils {
   def copyLogicalPlanWithUpdatedTimeRange(logicalPlan: LogicalPlan,
                                           timeRange: TimeRange): LogicalPlan = {
     logicalPlan match {
-      case lp: PeriodicSeriesPlan => copyWithUpdatedTimeRange(lp, timeRange)
-      case lp: RawSeriesLikePlan => copyNonPeriodicWithUpdatedTimeRange(lp, timeRange)
-      case _ => throw new UnsupportedOperationException("Logical plan not supported for copy")
+      case lp: PeriodicSeriesPlan  => copyWithUpdatedTimeRange(lp, timeRange)
+      case lp: RawSeriesLikePlan   => copyNonPeriodicWithUpdatedTimeRange(lp, timeRange)
+      case lp: LabelValues         => lp.copy(startMs = timeRange.startMs, endMs = timeRange.endMs)
+      case lp: SeriesKeysByFilters => lp.copy(startMs = timeRange.startMs, endMs = timeRange.endMs)
```

Comment:
confirmed that when one of these is omitted, it results in compile error ? (because of sealed class)

Reply:
Yes it throws `match may not be exhaustive` error as LogicalPlan is a sealed class